src/app/client/src/app/modules/shared/components/redirect/redirect.component.spec.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  ResourceService,
+  ConfigService,
+  BrowserCacheTtlService
+} from '@sunbird/shared';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { RedirectComponent } from './redirect.component';
+import { ActivatedRoute, RouterModule, Router } from '@angular/router';
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { TelemetryModule } from '@sunbird/telemetry';
+import { Observable } from 'rxjs/Observable';
+import { RouterTestingModule } from '@angular/router/testing';
+import { CacheService } from 'ng2-cache-service';
+
+describe('RedirectComponent', () => {
+  let component: RedirectComponent;
+  let fixture: ComponentFixture<RedirectComponent>;
+  const fakeActivatedRoute = {
+    snapshot: {
+      data: {
+        telemetry: {
+          env: 'redirect',
+          pageid: 'learn-redirect',
+          type: 'view',
+          object: { type: '', ver: '1.0' }
+        }
+      }
+    }
+  };
+
+  class RouterStub {
+    navigate = jasmine.createSpy('navigate');
+  }
+
+  beforeEach(async(() => {
```

Comment:
why is there beforeEach twice and what is the use of setting this in beforeEach
can't we use beforeAll please explain

Reply:
Beforeall is failing in our case, so i am using beforeEach itself